src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData-CSharp.Web/Program.cs

```diff
@@ -92,10 +94,14 @@
         new AzureKeyCredential(builder.Configuration["AzureAISearch:Key"] ?? throw new InvalidOperationException("Missing configuration: AzureAISearch:Key. See the README for details."))));
 #endif
 #else // UseLocalVectorStore
-var vectorStore = new JsonVectorStore(Path.Combine(AppContext.BaseDirectory, "vector-store"));
+var vectorStorePath = Path.Combine(AppContext.BaseDirectory, "vector-store.db");
+var vectorStore = new SqliteVectorStore($"Data Source={vectorStorePath}", new SqliteVectorStoreOptions()
+{
+    EmbeddingGenerator = embeddingGenerator,
+});
 #endif
 
-builder.Services.AddSingleton<IVectorStore>(vectorStore);
+builder.Services.AddSingleton<VectorStore>(vectorStore);
```

Comment:
Just want to make sure that the DI registration method (e.g. AddSqliteVectorStore) is intentionally not being used here - I see AddChatClient() below, are we avoiding the MEVD DI method for some reason?

Reply:
Main reason was to keep the template changes small (since the Azure AI Search template code also didn't use its corresponding registration method), but it actually simplifies the template a bit to update all vector store registrations, so I've just done that 🙂